### PR TITLE
Fix Michigan home heating credit

### LINKED ISSUE #3585: Fix Michigan home heating credit
In the 0.615.0 version, the MI home heating credit amount is incorrectly calculated in some cases.

For example, an elderly couple with $12,020 income and four dependents gets a credit of over **five million dollars**.  For the same case, TAXSIM35 generates a heating credit of less than one thousand dollars.   

These results indicate that the project's stated goal of using "test-driven development" is not being practiced at an adequate level.  By any sensible definition of test-driven development, the pull request that added the MI heating credit should have been able to detect an error of this magnitude.

### PR DESCRIPTION
**Pull request by Mshuning**
Fixes #3585

### CODE REVIEW

File: policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  output: 
+    mi_alternate_home_heating_credit_eligible: true
+    # (1_000 - 12_020 * 0.11) * 0.7 < 0
+    mi_alternate_home_heating_credit: 0
+    mi_standard_home_heating_credit_eligible: true
+    # 1433 - 12_020 * 0.035
+    mi_standard_home_heating_credit: 1012.3
+    mi_home_heating_credit_eligible_rate: 1
+    # 1012.3 * 0.9
+    mi_home_heating_credit: 911.07
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml:76]: add a test 

- name: integration test 2, an elderly couple with $12,020 income and four dependents
  period: 2022
  input:
    people:
      person1:
        age: 65
        employment_income: 6_000
      person2:
        age: 65
        employment_income: 6_020
      person3:
        age: 10
        employment_income: 0
      person4:
        age: 8
        employment_income: 0
      person5:
        age: 6
        employment_income: 0
      person6:
        age: 4
        employment_income: 0
    tax_units:
      tax_unit:
        # heating_expenses: 1_000
        # utilities_included_in_rent: false
        members: [person1, person2, person3, person4]
    households:
      household:
        members: [person1, person2, person3, person4]
        state_code: MI
  output: 
    mi_alternate_home_heating_credit_eligible: true
    # (1_000 - 12_020 * 0.11) * 0.7 < 0
    mi_alternate_home_heating_credit: 0
    mi_standard_home_heating_credit_eligible: true
    # 1433 - 12_020 * 0.035
    mi_standard_home_heating_credit: 1012.3
    mi_home_heating_credit_eligible_rate: 1
    # 1012.3 * 0.9
    mi_home_heating_credit: 911.07

File: policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    # (1_000 - 1_000 * 0.11) * 0.7
+    mi_alternate_home_heating_credit: 623
+    mi_standard_home_heating_credit_eligible: true
+    # 706 - 1000 * 0.035
+    mi_standard_home_heating_credit: 671
+    mi_home_heating_credit_eligible_rate: 1
+    # 671 * 0.9
+    mi_home_heating_credit: 603.9
+
+- name: integretion test 2, an elderly couple with $12,020 income and four dependents
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml:38]:
```suggestion
- name: integration test 2, an elderly couple with $12,020 income and four dependents
```

**Comment by PavelMakarchuk** [comment]
> @PavelMakarchuk To fix the bug, I have to create a new file related to `mi_exemptions`, which is the number of people eligible for MI exemptions. But it will not be involved in the calculation of `mi_exemptions` itself, since different types of eligible people share different exemption amount. Should I create this new variable in `exemptions` folder? image

We need to total number (int) of exemptions right? Not the total value of the exemptions (the dollar amount that you get per exemption)? If we need the number then we need to create a `mi_exemptions_count.py` and if we need the value we can take `mi_exemptions`

File: policyengine_us/variables/gov/states/mi/tax/income/exemptions/mi_exemptions_count.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    defined_for = StateCode.MI
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "http://legislature.mi.gov/doc.aspx?mcl-206-30",
+        "https://www.legislature.mi.gov/Publications/TaxpayerGuide.pdf",
+        "https://www.michigan.gov/-/media/Project/Websites/taxes/2022RM/IIT/MI-1040CR7.pdf",
+        "https://www.michigan.gov/-/media/Project/Websites/taxes/2022RM/IIT/BOOK_MI-1040CR-7.pdf#page=7",
+    )
+
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mi/tax/income/exemptions/mi_exemptions_count.py:17]: Would this work?
```suggestion
    adds = ["tax_unit_size", "mi_disabled_exemption_eligible_person", "is_fully_disabled_service_connected_veteran"]
```

File: policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        claimed_as_dependent_on_another_return: false
+    tax_units:
+      tax_unit:
+        mi_household_resources: 1_000
+        mi_exemptions_count: 2
+        heating_expenses: 1_000
+        tax_unit_fpg: 23_030
+        utilities_included_in_rent: false
+        members: [person1, person2, person3]
+        tax_unit_size: 3
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/mi/tax/income/credits/home_heating/integration.yaml:22]: this and `mi_exemptions_count` are inconsistent. they are also derived from the person-level information so neither is needed. in general for integration tests we should provide only "leaf-node" inputs - those without a formula. so also e.g. please remove tax_unit_fpg and mi_household_resources

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**